common/browser.go

```diff
@@ -123,6 +123,9 @@ func (b *Browser) connect() error {
 
 	b.conn = conn
 
+	// Start the connection to listen for CDP events.
+	conn.start()
```

Comment:
I'll answer your other comments, but I have a partial answer for this one: https://github.com/grafana/xk6-browser/pull/1219#discussion_r1497782514. `start()` controls and makes it explicit when to start `Connection`'s background goroutines.

> Could the hook be passed into the NewConnection as an argument, such as:

Yes, but then all the tests would need to be updated with `nil, ' which I didn't prefer to do so because they don't need this hook to work. I can change this part if you think it's important.

Reply:
> I have a partial answer for this one also here: https://github.com/grafana/xk6-browser/pull/1219#discussion_r1497782514. Using start(), we can explicitly control when to start Connection's background goroutines.

We can discuss that in the other comment.

> Yes, but then all the tests would need to be updated with `nil, ' which I didn't prefer to do so because they don't need this hook to work. I can change this part if you think it's important.

If that's the only reason then my preference is to pass it as an argument in `NewConnection`. Otherwise it feels like a hidden feature.

Reply:
See 1746b073.

Reply:
See e0d7db06